docs/components/component-lifecycle.md

```diff
@@ -13,8 +13,9 @@ Components have numerous lifecycle methods which can be used to know when the co
 
 Implement one of the following methods within a component class and Stencil will automatically call them in the right order:
 
-<lifecycle-chart></lifecycle-chart>
+import LifecycleMethodsChart from '@site/src/components/LifecycleMethodsChart';
 
+<LifecycleMethodsChart />
```

Comment:
because these files are processed by Docusaurus as 'mdx' files we can just import a React component and render them as if we were in a `.jsx` or `.tsx` file (we don't need to `import React from 'react'` however)

I believe that there are some differences between the `md` and `mdx` parsers, and if you have anything it believes is JSX it will switch to using the `mdx` one. You can also force this by using an `.mdx` file extension.

More info here: https://docusaurus.io/docs/markdown-features/react